app/src/org/commcare/activities/connect/HQUserInviteActivity.java

```diff
@@ -0,0 +1,173 @@
+    String connectUserName;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        Logger.log("HQInvite", "Entering HQ user invite activity");
+
+        binding = ActivityHquserInviteBinding.inflate(getLayoutInflater());
+        setContentView(binding.getRoot());
+
+        Intent intent = getIntent();
+        Uri data = intent.getData();
+        if (data != null) {
+            callBackURL = data.getQueryParameter("callback_url");
+            username = data.getQueryParameter("hq_username");
+            inviteCode = data.getQueryParameter("invite_code");
+            domain = data.getQueryParameter("hq_domain");
+            connectUserName = data.getQueryParameter("connect_username");
+        }
+
+        FirebaseAnalyticsUtil.reportHQInvitationDeepLink(domain);
+
+        handleButtons();
+    }
+
+    private void handleButtons() {
+        ConnectManager.init(this);
+
+        ConnectUserRecord user = ConnectManager.getUser(this);
+        boolean isTokenPresent = ConnectManager.isConnectIdConfigured();
+        boolean isCorrectUser = user == null || user.getUserId().equals(connectUserName);
+
+        if (isCorrectUser) {
+            binding.tvHqInvitationHeaderTitle.setText(isTokenPresent
+                    ? getString(R.string.connect_hq_invitation_heading, username)
+                    : getString(R.string.connect_hq_invitation_connectId_not_configure));
+            setButtonVisibility(isTokenPresent);
+            setButtonListeners(isTokenPresent);
+        } else {
+            binding.tvHqInvitationHeaderTitle.setText(getString(R.string.connect_hq_invitation_wrong_user));
+            setButtonVisibility(false);
+        }
+    }
+
+    private void setButtonVisibility(boolean visible) {
+        binding.btnAcceptInvitation.setVisibility(visible ? View.VISIBLE : View.GONE);
+        binding.btnDeniedInvitation.setVisibility(visible ? View.VISIBLE : View.GONE);
+        binding.btnGoToRecovery.setVisibility(visible ? View.GONE : View.VISIBLE);
+    }
+
+    private void setButtonListeners(boolean isTokenPresent) {
+        if (isTokenPresent) {
+            binding.btnAcceptInvitation.setOnClickListener(view -> handleInvitation(callBackURL, inviteCode));
+            binding.btnDeniedInvitation.setOnClickListener(view -> declineInvitation());
+        } else {
+            binding.btnGoToRecovery.setOnClickListener(view -> ConnectManager.registerUser(this, success -> {
+                        if (success) {
+                            ConnectManager.goToConnectJobsList(this);
+                        }
+                    })
+            );
+        }
+    }
+
+    private void declineInvitation() {
+        FirebaseAnalyticsUtil.reportHQInvitationResponse(domain, false, "");
+        finish();
+    }
+
+    private void handleInvitation(String callBackUrl, String inviteCode) {
+        Logger.log("HQInvite", "User accepted invitation");
```

Comment:
Use constants for Log Types otherwise we risk having a lot of different scattered categories in logs.  You can add a connect specific class to signify logtypes like `ConnectLogTypes` if it's easier.